src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1388,6 +1373,21 @@ int PeerReplayer::do_synchronize(const std::string &dir_root, const Snapshot &cu
 
     sync_stack.pop();
   }
+end:
+  dout(20) << " current:" << (close_current ? fh.c_fd : -1)
+           << " prev:" << fh.p_fd
+           << " ret = " << r
+           << dendl;
+
+  // @FHandles.r_fd_dir_root is closed in @unregister_directory since
+  // its used to acquire an exclusive lock on remote dir_root.
+
+  // if c_fd has been used in ceph_fdopendir call then
+  // there is no need to clost this fd manually.
+  if (close_current) {
+    ceph_close(m_local_mount, fh.c_fd);
```

Comment:
I think we need to use `ceph_closedir` here instead, which uses `dir_result_t`.

Reply:
Shouldn't this ceph_closedir already closing fh.c_fd if it reaches there while popping the stack? In that case shouldn't be the ceph_close call redundent?

Reply:
Okay, but yeah, my thought was regarding if ceph_closedir fails.

Reply:
ceph_closedir() has the only way to return an error - when cmount isn't mounted, nad Client::closedir always returns 0:

extern "C" int ceph_closedir(struct ceph_mount_info *cmount, struct ceph_dir_result *dirp)
{
  if (!cmount->is_mounted())
    return -CEPHFS_ENOTCONN;
  return cmount->get_client()->closedir(reinterpret_cast<dir_result_t*>(dirp));
}

So we might want to just put ceph_assert() here to make sure we never getting the error..